activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -822,6 +832,21 @@ def check_constraint(*args)
       def remove_check_constraint(*args)
         @base.remove_check_constraint(name, *args)
       end
+
+      private
+        def raise_on_if_exist_options(options)
+          unrecognized_option = options.keys.find do |key|
+            key == :if_exists || key == :if_not_exists
+          end
+          if unrecognized_option
+            conditional = unrecognized_option == :if_exists ? "if" : "unless"
+            message =
+              "Option #{unrecognized_option} will be ignored. If you are calling an expression like " +
+              "`t.column(.., #{unrecognized_option}: true)` from inside a change_table block, " +
```

Comment:
I think we know that's what they're calling?

(I think we could probably also pull the `t.column` out of `caller`, if we want to be really specific about what they want.)

Reply:
Come to think of it, "is not supported" might be more true than "will be ignored", given we're erroring when it's present. Prior to this change it _would've_ been ignored, but that's not true in the post-this-PR state.

Reply:
I'm not sure we can know with certainty how the written line reads (whether, for instance, it was `t.column(:name, :string, ...)` or `t.string(:name, ...)`, which is why I kept it generic. Would it be confusing if the error message was correct about the name and type of the column, but wrong about the method invocation? I'll change it around if not.

Reply:
Ah, fair. Yeah I'd say if you pull it from caller, we'll be right for anything that isn't a column definition, and for `t.string` we'll be saying `t.column`, which is [half-true, and] the same as what we're always doing now. So probably fine?

Reply:
Oh I see what you mean by pulling it out of caller. I could do that with, say, this expression `method_used = caller.first[/(?<=:in `)[^']+(?=')/] || "column"` but I think faithfully reproducing the original written line of code would be fraught when you consider edge cases like `remove`, which accepts multiple column names, or `timestamps`, which doesn't take any.